resolwe_bio/processes/reads_processing/cutadapt.yml

```diff
@@ -396,82 +406,55 @@
-        {% if error_rate %}--error-rate {{ error_rate }} {% endif %} \
-        -O {{min_overlap_fwd}} \
-        -o "${NAME1}_trimmed.fastq.gz" in_fw_reads.fastq.gz > report.txt
-      re-checkrc "Failed while running cutadapt"
-      re-progress 0.4
-
-      # Run Cutadapt on reverse reads
-      cutadapt \
-        {% if headcrop_rev %}-u {{headcrop_rev}}{% endif %} \
-        {% if crop_rev %}-u -{{crop_rev}}{% endif %} \
-        {% if leading %}-q {{leading}}{% endif %} \
-        {% if trailing %}-q {{trailing}},0{% endif %} \
-        {% if polya_tail %}-a A{$polya_tail}N{1}{% endif %} \
-        {% if down_primers_seq_rev %}$down_primers_rev{% endif %} \
-        {% if up_primers_seq_rev %}$up_primers_rev{% endif %} \
-        {% if down_primers_file_rev %}-a file:3_primers_rev.fasta{% endif %} \
-        {% if up_primers_file_rev %}-g file:5_primers_rev.fasta{% endif %} \
-        {% if times %}--times {{ times }} {% endif %} \
-        {% if match_read_wildcards %}--match-read-wildcards {% endif %} \
-        {% if error_rate %}--error-rate {{ error_rate }} {% endif %} \
-        -O {{min_overlap_rev}} \
-        -o "${NAME2}_trimmed.fastq.gz" in_rw_reads.fastq.gz >> report.txt
-      re-checkrc "Failed while running cutadapt"
-      re-progress 0.7
-
-      # Run Cutadapt on both reads
-      {% if minlen %}
-        if [ -f "${NAME1}_trimmed.fastq.gz" ]; then
-         mv "${NAME1}_trimmed.fastq.gz" in_fw_reads.fastq.gz
-        fi
-        if [ -f "${NAME2}_trimmed.fastq.gz" ]; then
-         mv "${NAME2}_trimmed.fastq.gz" in_rw_reads.fastq.gz
-        fi
-        cutadapt -m 20 -o "${NAME1}_trimmed.fastq.gz" -p "${NAME2}_trimmed.fastq.gz" in_fw_reads.fastq.gz in_rw_reads.fastq.gz >> report.txt
-      {% endif %}
+        {% if filtering.minlen %}--minimum-length={{ filtering.minlen }}{% endif %} \
+        {% if modify_reads.headcrop %}--cut={{ modify_reads.headcrop }}{% endif %} \
+        {% if modify_reads.crop %}--cut=-{{ modify_reads.crop }}{% endif %} \
+        {% if modify_reads.trailing %}--quality-cutoff={{ modify_reads.trailing }}{% endif %} \
+        {% if modify_reads.leading %}--quality-cutoff={{ modify_reads.leading }},0{% endif %} \
+        {% if adapters.down_primers_seq_fwd %}$down_primers_fwd{% endif %} \
+        {% if adapters.up_primers_seq_fwd %}$up_primers_fwd{% endif %} \
+        {% if adapters.down_primers_file_fwd %}-a file:3_primers_fwd.fasta{% endif %} \
+        {% if adapters.up_primers_file_fwd %}-g file:5_primers_fwd.fasta{% endif %} \
+        {% if adapters.down_primers_seq_rev %}$down_primers_rev{% endif %} \
+        {% if adapters.up_primers_seq_rev %}$up_primers_rev{% endif %} \
+        {% if adapters.down_primers_file_rev %}-A file:3_primers_rev.fasta{% endif %} \
+        {% if adapters.up_primers_file_rev %}-G file:5_primers_rev.fasta{% endif %} \
+        {% if adapters.match_read_wildcards %}--match-read-wildcards {% endif %} \
```

Comment:
This parameter is not optional and has a default value. Remove the if-cluase and reformat into form similar to `--pair-filter` / `--error-rate`, `--times`, ...

Reply:
because the `type: basic:boolean` it should stay like it is.